Godeps/_workspace/src/github.com/samuel/go-zookeeper/zk/conn.go

```diff
@@ -454,9 +454,6 @@ func (c *Conn) authenticate() error {
 		return ErrSessionExpired
 	}
 
-	if c.sessionID != r.SessionID {
-		atomic.StoreUint32(&c.xid, 0)
-	}
```

Comment:
At the time when I created this PR my changes have not yet been accepted to the upstream master. Although now they are, and we can switch to it.